libnavigator/src/main/java/com/mapbox/navigation/navigator/internal/NavigatorLoader.kt

```diff
@@ -27,17 +30,27 @@ internal object NavigatorLoader {
     fun createNavigator(
         deviceProfile: DeviceProfile,
         navigatorConfig: NavigatorConfig,
-        tilesConfig: TilesConfig
+        tilesConfig: TilesConfig,
+        historyDir: String?
     ): NativeComponents {
         val config = ConfigFactory.build(
             settingsProfile(deviceProfile),
             navigatorConfig,
             deviceProfile.customConfig
         )
         val runLoopExecutor = RunLoopExecutorFactory.build()
-        val historyRecorder = null
-        val cache = CacheFactory.build(tilesConfig, config, historyRecorder)
 
+        val historyRecorder = if (historyDir != null) {
+            HistoryRecorderHandle.build(historyDir, config)
```

Comment:
Noting that `HistoryRecorderHandle.build` may return `null` if the `historyDir` is misconfigured or not accessible 👀 

https://github.com/mapbox/mapbox-navigation-android/pull/4478/commits/fff52fecadb49e49de167a96a9bc2c7782dc5c80#diff-bfb34db9034b019b28925420b11d6d1ab67e9b5c4ea973fc805904c2b812f505L43-L52

Reply:
This is okay, this is why the HistoryRecorderController throws a warning rather than an exception

``` kotlin
            historyRecorderHandle?.apply {
                dumpHistory { filePath: String? ->
                    result.onSaved(filePath)
                }
            } ?: run {
                logger.w(
                    Tag("MbxHistoryRecorderOptions"),
                    Message("dumpHistory failed - The history recorder is not initialized")
                )
                result.onSaved(null)
            }
```

Reply:
My point is that there are different reasons why the `historyRecorder` may be `null`:

1. `historyDir` not configured at all
2. `historyDir` configured but `HistoryRecorderHandle.build(historyDir, config)` returns `null`

Do we want to log each scenario?

Reply:
This is still relevant.

Reply:
> 1. historyDir not configured at all
> 2. historyDir configured but HistoryRecorderHandle.build(historyDir, config) returns null

When it is not configured, an automatic location is created. in `mbx_nav/history`

The configuration is deep in `NavigatorLoader`. 
``` kotlin
        val historyRecorder = if (historyDir != null) {
            HistoryRecorderHandle.build(historyDir, config)
        } else {
            null
        }
```

The log falls downstream, and is captured by `dumpHistory failed - The history recorder is not initialized`

So both of the cases are captured by the same log. I figured that is granular enough for now

Reply:
added logs in NatigatorLoader